components/dialog/dialog-mixin.js

```diff
@@ -315,10 +316,19 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		// native dialog backdrop does not prevent body scrolling
 		this._bodyScrollKey = preventBodyScroll();
 
+		// focus first focusable child prior to auto resize (fixes screen reader hiccups)
+		this._focusInitial();
+
 		requestAnimationFrame(async() => {
 			await this._updateSize();
 			this._state = 'showing';
-			this._focusInitial();
+
+			// edge case: no children were selected, try again after one redraw
+			if (!dialog.contains(document.activeElement)
+			&& !(footer && footer.assignedNodes().includes(document.activeElement))) {
+				this._focusInitial();
+			}
```

Comment:
Are you finding that `dialog.contains` is working with shadow roots? Maybe the fact that `document.activeElement` is going to hide the shadow roots is making this work. Curious what it's returning actually.

We have a helper that might make this easier in `helpers/focus.js` called `getComposedActiveElement()` which will return that actual composed element (the `<button>` element inside `d2l-button` inside the footer). You could then check that it's a descendant of the dialog using `isComposedAncestor` helper in `helpers/dom.js`.

Reply:
In this case `document.activeElement` is evaluating to the `d2l-button` component instead of the actual composed element, which is why the contains check fails and I assumed the footer slot check would be necessary - though @dbatiste is right about that being less future-proof.

Reply:
Cool yeah, if this were wrapping in another shadowDOM `document.activeElement` would resolve to the element in the outer context, so it all depends on the context in which is it used.